### ISSUE TITLE
Service is not listening location changes, not in app , not in foreground, not on terminated

### ISSUE THREAD

**Issue by mahirozdin** [transistorsoft/flutter_background_geolocation#1046]
## Your Environment
* Plugin version: 4.11.0
* Platform: Android
* OS version: API 33 
* Device manufacturer / model: Samsung Note 20 ULTRA / Several Devices / Emulator
* Flutter info (`flutter doctor`):
```
[√] Flutter (Channel stable, 3.3.9, on Microsoft Windows [Version 10.0.22621.1555], locale tr-TR)
[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[√] Chrome - develop for the web
[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.3.3)
[√] Android Studio (version 2022.2)
[√] VS Code (version 1.78.2)
[√] Connected device (4 available)
[√] HTTP Host Availability
```
* Plugin config:
```dart 
bg.Config(
        
        debug: true,
        // useSignificantChangesOnly: , //? This can be a solution for ios?

        preventSuspend: true,
        heartbeatInterval: 9000,
        desiredAccuracy: bg.Config.DESIRED_ACCURACY_HIGH,
        distanceFilter: 100.0,
        geofenceInitialTriggerEntry: true,
        enableHeadless: true,
        stopOnTerminate: false,
        disableStopDetection: true,
        pausesLocationUpdatesAutomatically: false,
        logLevel: bg.Config.LOG_LEVEL_VERBOSE,
        startOnBoot: true,
      )
```

## Expected Behavior
It was working as its expected. But somehow for 3 days its not working. Normaly when it starts there is a debig sound like ciu-ciu-ciu like lazer gun voice was coming and it was working as expected. Now its not. Without changing anything im restarting 10 times on one of them that voice is coming and when that voice comes its works as expected. 

Before the problem It was working like magic in foregrand backgground terminated and while app open but now none of them happenss. 

## Actual Behavior
Every time when app opens it should work or that voice should come

## Steps to Reproduce
just restarting a few times.

## Context
Im coding a family locator app. Background tasks and in app listeners sends data to firebase. And it was working without a problem. 

## Debug logs

<details>
	<summary>Logs</summary>

``` <!-- syntax-highligting:  DO NOT REMOVE -->
05-17 12:11:43.928 10612 10733 D TSLocationManager: [c.t.l.logger.LoggerFacade$a a] com.transistorsoft/flutter_background_geolocation/events/location
05-17 12:11:43.929 10612 10733 D TSLocationManager: [c.t.l.logger.LoggerFacade$a a] com.transistorsoft/flutter_background_geolocation/events/motionchange
05-17 12:11:43.929 10612 10733 D TSLocationManager: [c.t.l.logger.LoggerFacade$a a] com.transistorsoft/flutter_background_geolocation/events/activitychange
05-17 12:11:43.929 10612 10733 D TSLocationManager: [c.t.l.logger.LoggerFacade$a a] com.transistorsoft/flutter_background_geolocation/events/geofenceschange
05-17 12:11:43.929 10612 10733 D TSLocationManager: [c.t.l.logger.LoggerFacade$a a] com.transistorsoft/flutter_background_geolocation/events/geofence
05-17 12:11:43.930 10612 10733 D TSLocationManager: [c.t.l.logger.LoggerFacade$a a] com.transistorsoft/flutter_background_geolocation/events/heartbeat
05-17 12:11:43.930 10612 10733 D TSLocationManager: [c.t.l.logger.LoggerFacade$a a] com.transistorsoft/flutter_background_geolocation/events/http
05-17 12:11:43.930 10612 10733 D TSLocationManager: [c.t.l.logger.LoggerFacade$a a] com.transistorsoft/flutter_background_geolocation/events/schedule
05-17 12:11:43.930 10612 10733 D TSLocationManager: [c.t.l.logger.LoggerFacade$a a] com.transistorsoft/flutter_background_geolocation/events/connectivitychange
05-17 12:11:43.930 10612 10733 D TSLocationManager: [c.t.l.logger.LoggerFacade$a a] com.transistorsoft/flutter_background_geolocation/events/enabledchange
05-17 12:11:43.930 10612 10733 D TSLocationManager: [c.t.l.logger.LoggerFacade$a a] com.transistorsoft/flutter_background_geolocation/events/providerchange
05-17 12:11:43.930 10612 10733 D TSLocationManager: [c.t.l.logger.LoggerFacade$a a] com.transistorsoft/flutter_background_geolocation/events/powersavechange
05-17 12:11:43.930 10612 10733 D TSLocationManager: [c.t.l.logger.LoggerFacade$a a] com.transistorsoft/flutter_background_geolocation/events/notificationaction
05-17 12:11:43.930 10612 10733 D TSLocationManager: [c.t.l.logger.LoggerFacade$a a] com.transistorsoft/flutter_background_geolocation/events/authorization
05-17 12:11:43.931 10612 10733 I TSLocationManager: [c.t.l.logger.LoggerFacade$a a]
05-17 12:11:43.931 10612 10733 I TSLocationManager:   ✅  Google Play Services: connected (version code:12451000)
05-17 12:11:43.931 10612 10733 D TSLocationManager: [c.t.l.logger.LoggerFacade$a a] PID: 3
05-17 12:11:43.931 10612 10733 I TSLocationManager: [c.t.l.logger.LoggerFacade$a a]
05-17 12:11:43.931 10612 10733 I TSLocationManager: ╔═════════════════════════════════════════════
05-17 12:11:43.931 10612 10733 I TSLocationManager: ║ TSLocationManager version: 3.4.0 (420)
05-17 12:11:43.931 10612 10733 I TSLocationManager: ╠═════════════════════════════════════════════
05-17 12:11:43.931 10612 10733 I TSLocationManager: ╟─ samsung SM-N986B @ 13 (flutter)
05-17 12:11:43.931 10612 10733 I TSLocationManager: {
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "activityRecognitionInterval": 10000,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "allowIdenticalLocations": false,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "authorization": {},
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "autoSync": true,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "autoSyncThreshold": 0,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "backgroundPermissionRationale": {
05-17 12:11:43.931 10612 10733 I TSLocationManager:     "title": "Allow {applicationName} to access this device's location even when closed or not in use?",
05-17 12:11:43.931 10612 10733 I TSLocationManager:     "message": "[CHANGEME] This app collects location data for FEATURE X and FEATURE Y.",
05-17 12:11:43.931 10612 10733 I TSLocationManager:     "positiveAction": "Change to \"{backgroundPermissionOptionLabel}\"",
05-17 12:11:43.931 10612 10733 I TSLocationManager:     "negativeAction": ""
05-17 12:11:43.931 10612 10733 I TSLocationManager:   },
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "batchSync": false,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "configUrl": "",
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "debug": true,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "deferTime": 0,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "desiredAccuracy": -1,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "desiredOdometerAccuracy": 100,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "disableAutoSyncOnCellular": false,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "disableElasticity": false,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "disableLocationAuthorizationAlert": false,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "disableMotionActivityUpdates": false,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "disableProviderChangeRecord": false,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "disableStopDetection": true,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "distanceFilter": 100,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "elasticityMultiplier": 1,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "enableHeadless": true,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "enableTimestampMeta": false,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "extras": {},
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "fastestLocationUpdateInterval": -1,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "foregroundService": true,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "geofenceInitialTriggerEntry": true,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "geofenceModeHighAccuracy": false,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "geofenceProximityRadius": 1000,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "geofenceTemplate": "",
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "headers": {},
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "headlessJobService": "com.transistorsoft.flutter.backgroundgeolocation.HeadlessTask",
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "heartbeatInterval": 9000,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "httpRootProperty": "location",
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "httpTimeout": 60000,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "isMoving": false,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "locationAuthorizationRequest": "Always",
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "locationTemplate": "",
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "locationTimeout": 60,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "locationUpdateInterval": 1000,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "locationsOrderDirection": "ASC",
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "logLevel": 5,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "logMaxDays": 3,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "maxBatchSize": -1,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "maxDaysToPersist": 1,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "maxRecordsToPersist": -1,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "method": "POST",
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "minimumActivityRecognitionConfidence": 75,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "motionTriggerDelay": 0,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "notification": {
05-17 12:11:43.931 10612 10733 I TSLocationManager:     "layout": "",
05-17 12:11:43.931 10612 10733 I TSLocationManager:     "title": "",
05-17 12:11:43.931 10612 10733 I TSLocationManager:     "text": "Location Service activated",
05-17 12:11:43.931 10612 10733 I TSLocationManager:     "color": "",
05-17 12:11:43.931 10612 10733 I TSLocationManager:     "channelName": "TSLocationManager",
05-17 12:11:43.931 10612 10733 I TSLocationManager:     "channelId": "",
05-17 12:11:43.931 10612 10733 I TSLocationManager:     "smallIcon": "",
05-17 12:11:43.931 10612 10733 I TSLocationManager:     "largeIcon": "",
05-17 12:11:43.931 10612 10733 I TSLocationManager:     "priority": 0,
05-17 12:11:43.931 10612 10733 I TSLocationManager:     "sticky": false,
05-17 12:11:43.931 10612 10733 I TSLocationManager:     "strings": {},
05-17 12:11:43.931 10612 10733 I TSLocationManager:     "actions": []
05-17 12:11:43.931 10612 10733 I TSLocationManager:   },
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "params": {},
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "persist": true,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "persistMode": 2,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "schedule": [],
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "scheduleUseAlarmManager": false,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "speedJumpFilter": 300,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "startOnBoot": true,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "stationaryRadius": 25,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "stopAfterElapsedMinutes": 0,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "stopOnStationary": false,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "stopOnTerminate": false,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "stopTimeout": 5,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "triggerActivities": "in_vehicle, on_bicycle, on_foot, running, walking",
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "url": "",
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "useSignificantChangesOnly": false,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "enabled": false,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "schedulerEnabled": false,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "trackingMode": 1,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "odometer": 0,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "isFirstBoot": false,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "didLaunchInBackground": false,
05-17 12:11:43.931 10612 10733 I TSLocationManager:   "didDeviceReboot": false
05-17 12:11:43.931 10612 10733 I TSLocationManager: }
05-17 12:11:43.931 10612 10733 I TSLocationManager: [c.t.l.logger.LoggerFacade$a a]
05-17 12:11:43.931 10612 10733 I TSLocationManager: ╔═════════════════════════════════════════════
05-17 12:11:43.931 10612 10733 I TSLocationManager: ║ DEVICE SENSORS
05-17 12:11:43.931 10612 10733 I TSLocationManager: ╠═════════════════════════════════════════════
05-17 12:11:43.931 10612 10733 I TSLocationManager: ╟─ ✅  ACCELEROMETER: {Sensor name="LSM6DSO Accelerometer", vendor="STM", version=15932, type=1, maxRange=78.4532, resolution=0.0023942017, power=0.2, minDelay=5000}
05-17 12:11:43.931 10612 10733 I TSLocationManager: ╟─ ✅  GYROSCOPE: {Sensor name="LSM6DSO Gyroscope", vendor="STM", version=1, type=4, maxRange=17.453032, resolution=6.1086525E-4, power=0.3, minDelay=5000}
05-17 12:11:43.931 10612 10733 I TSLocationManager: ╟─ ✅  MAGNETOMETER: {Sensor name="AK09918C Magnetometer", vendor="AKM", version=1, type=2, maxRange=1999.98, resolution=0.06, power=0.6, minDelay=10000}
05-17 12:11:43.931 10612 10733 I TSLocationManager: ╟─ ✅  SIGNIFICANT_MOTION: {Sensor name="Significant Motion Sensor", vendor="Samsung Electronics", version=2, type=17, maxRange=1.0, resolution=1.0, power=0.3, minDelay=-1}
05-17 12:11:43.931 10612 10733 I TSLocationManager: ╚═════════════════════════════════════════════
05-17 12:11:43.931 10612 10733 D TSLocationManager: [c.t.l.logger.LoggerFacade$a a] ☯️  onCreate
05-17 12:11:43.931 10612 10733 D TSLocationManager: [c.t.l.logger.LoggerFacade$a a] ☯️  onStart
05-17 12:11:43.931 10612 10733 D TSLocationManager: [c.t.l.logger.LoggerFacade$a a] ☯️  onResume
05-17 12:11:43.931 10612 10733 I TSLocationManager: [c.t.l.logger.LoggerFacade$a a]
05-17 12:11:43.931 10612 10733 I TSLocationManager:   🎾  Start monitoring location-provider changes
05-17 12:11:43.931 10612 10733 D TSLocationManager: [c.t.l.logger.LoggerFacade$a a]
05-17 12:11:43.931 10612 10733 D TSLocationManager:   🔴  Cleared callbacks
05-17 12:11:43.931 10612 10736 D TSLocationManager: [c.t.l.l.TSSQLiteAppender$c run]
05-17 12:11:43.931 10612 10736 D TSLocationManager:   ℹ️  Cleared logs older than 72 hours
05-17 12:11:43.931 10612 10733 D TSLocationManager: [c.t.l.logger.LoggerFacade$a a]
05-17 12:11:43.931 10612 10733 D TSLocationManager:   ℹ️  PRUNE -1 days
05-17 12:11:48.240 10612 10612 D TSLocationManager: [c.t.f.b.streams.StreamHandler onListen] location
05-17 12:11:48.242 10612 10612 D TSLocationManager: [c.t.f.b.streams.StreamHandler onListen] geofence
05-17 12:11:50.587 10612 10782 I TSLocationManager: [c.t.l.data.sqlite.GeofenceDAO destroyAll] ✅
05-17 12:11:51.106 10612 10782 I TSLocationManager: [c.t.l.data.sqlite.GeofenceDAO create]
05-17 12:11:51.106 10612 10782 I TSLocationManager:   ✅  JO1Ei4nvR0VI8eUBS9DK:9Hxbo19dRmRb8JJp2gBY
05-17 12:11:51.110 10612 10782 D TSLocationManager: [c.t.l.g.TSGeofenceManager d] ℹ️  Persist monitored geofences: []
05-17 12:11:51.113 10612 10696 I TSLocationManager: [c.t.l.data.sqlite.GeofenceDAO create]
05-17 12:11:51.113 10612 10696 I TSLocationManager:   ✅  JO1Ei4nvR0VI8eUBS9DK:DWi2xPImWKlrZMdXT9le
05-17 12:11:51.115 10612 10696 D TSLocationManager: [c.t.l.g.TSGeofenceManager d] ℹ️  Persist monitored geofences: []
05-17 12:11:51.116 10612 10782 I TSLocationManager: [c.t.l.data.sqlite.GeofenceDAO create]
05-17 12:11:51.116 10612 10782 I TSLocationManager:   ✅  JO1Ei4nvR0VI8eUBS9DK:Mc57cw5mW10MRLFrBy8J
05-17 12:11:51.118 10612 10782 D TSLocationManager: [c.t.l.g.TSGeofenceManager d] ℹ️  Persist monitored geofences: []
05-17 12:11:51.121 10612 10696 I TSLocationManager: [c.t.l.data.sqlite.GeofenceDAO create]
05-17 12:11:51.121 10612 10696 I TSLocationManager:   ✅  JO1Ei4nvR0VI8eUBS9DK:PTtq4G1mXwpVpMtsUzp3
05-17 12:11:51.123 10612 10696 D TSLocationManager: [c.t.l.g.TSGeofenceManager d] ℹ️  Persist monitored geofences: []
05-17 12:11:51.126 10612 10782 I TSLocationManager: [c.t.l.data.sqlite.GeofenceDAO create]
05-17 12:11:51.126 10612 10782 I TSLocationManager:   ✅  JO1Ei4nvR0VI8eUBS9DK:b3vyjua2vSMbZY2OMNlN
05-17 12:11:51.127 10612 10782 D TSLocationManager: [c.t.l.g.TSGeofenceManager d] ℹ️  Persist monitored geofences: []
05-17 12:11:51.129 10612 10696 I TSLocationManager: [c.t.l.data.sqlite.GeofenceDAO create]
05-17 12:11:51.129 10612 10696 I TSLocationManager:   ✅  JO1Ei4nvR0VI8eUBS9DK:rpb3dVtn0XDLvK1lam9D
05-17 12:11:51.131 10612 10696 D TSLocationManager: [c.t.l.g.TSGeofenceManager d] ℹ️  Persist monitored geofences: []
05-17 12:11:51.422 10612 10612 D TSLocationManager: [c.t.l.g.TSGeofenceManager$e run] evaluation buffer timer elapsed
05-17 12:11:51.428 10612 10733 E TSLocationManager: [c.t.l.l.TSLocationManager getLastLocation]
05-17 12:11:51.428 10612 10733 E TSLocationManager:   ‼️  Failed to get last location: com.transistorsoft.xms.g.tasks.Task$XImpl@6108889
```

</details>

**Comment by mahirozdin** [comment]
On this code state is getting false on each  restart

```dart
bg.BackgroundGeolocation.ready(bg.Config(
        
        debug: true,
        // useSignificantChangesOnly: , //? This can be a solution for ios?
        reset: true,
        preventSuspend: true,
        heartbeatInterval: 9000,
        desiredAccuracy: bg.Config.DESIRED_ACCURACY_HIGH,
        distanceFilter: 100.0,
        geofenceInitialTriggerEntry: true,
        enableHeadless: true,
        stopOnTerminate: false,
        disableStopDetection: true,
        pausesLocationUpdatesAutomatically: false,
        persistMode:  bg.Config.PERSIST_MODE_NONE,
        logLevel: bg.Config.LOG_LEVEL_VERBOSE,
        startOnBoot: true,
      )).then((bg.State state) {
        // clear all geofences

        if (state.enabled) {
          bg.BackgroundGeolocation.start();
          print("enabled");
        } else {
          print("not enabled");
        }
```

**Comment by christocracy** [comment]
You are telling the plug-in to .stop() somewhere.

**Comment by aweiand** [comment]
I have a problem similar to you. Aparently is related to app updates(?), when we send an update (Android/iOS or use in the emulator), sometimes the plugin don't register in android database (I'm whatching the app database under locations table) and don't send to server. I'm debug of many ways and cannot find the problem. 
But see well, when I wipe the emulator data and do a cold boot (without any code changes, cleans whatever....) works again!
Maybe, in my code I do many config changes over the day, it's necessary to call `.ready(....` all the time we change the config? or just `.setconfig(...` is enough?

Ps.: Sorry  for answering here and not in a new issue...

**Comment by christocracy** [comment]
The plugin will not initiate location tracking from simulated locations in emulator.  The plugin triggers location tracking when the device is *detected to be moving* (eg: Motion API).  The motion api doesn't work in simulator.

You will never experience the true behaviour of the plugin in the emulator.  You must test on a *real* device with *real* movement outside.

>it's necessary to call .ready(.... all the time we change the config? or just .setconfig(... is enough?

`.ready(config)` is called **ONLY ONCE** whenever your app boots.  To change the config *after* `.ready(config)` has been called, you use `.setConfig(newConfig)`.

**Comment by christocracy** [comment]
@aweiand I suggest you post a new issue and provide logs.  See wiki *Debugging*.

**Comment by mahirozdin** [comment]
Thanks for the reply. In my code there is no code such as `.stop()`. And its happining in the begining of the launch. As I told you  when we run plugin ready part It gives a state and state is return as false. 

Could you help me to understand when we run Ready part why state returns false. In which conditions state returns false. Maybe It could be related to my firebase firestore configuration. 

```dart
 FirebaseFirestore.instance.settings = Settings(persistenceEnabled: true); 
```

Do you think are there any conflicts related to this conf.

**Comment by christocracy** [comment]
Look at your **faulty logic** and say it out loud to yourself:

```dart
if (state.enabled) {
    bg.BackgroundGeolocation.start();
    print("enabled");
} else {
    print("not enabled");
}
```

Imagine if this were a light bulb and switch:   "*If the light bulb is ON, turn the switch ON*".

- Calling `.start()` sets `State.enabled -> true`
- Calling `.stop()` sets `State.enabled -> false`

It's that simple.  Like a light bulb and switch.

**Comment by mahirozdin** [comment]
Its not my faulty logic Its your documentation 

This code is taken by documentation 

```
bg.BackgroundGeolocation.ready(bg.Config(
        desiredAccuracy: bg.Config.DESIRED_ACCURACY_HIGH,
        distanceFilter: 10.0,
        stopOnTerminate: false,
        startOnBoot: true,
        debug: true,
        logLevel: bg.Config.LOG_LEVEL_VERBOSE
    )).then((bg.State state) {
      if (!state.enabled) {
        ////
        // 3.  Start the plugin.
        //
        bg.BackgroundGeolocation.start();
      }
    });
```

`bg.BackgroundGeolocation.ready` functions returns `bg.State state` and AS YOU TELL in your documentation if state is not false (so if it is true ) then `bg.BackgroundGeolocation.start()` 


So my code is 

```
if (state.enabled) {
    bg.BackgroundGeolocation.start();
    print("enabled");
} else {
    print("not enabled");
}
```

there is no diffrence between my code and documentation... 


as you tell if state gets enabled by calling ` bg.BackgroundGeolocation.start();` so it never gets true because you are running this code when state is true...

**Comment by christocracy** [comment]
Look at *your* code:
```dart
if (state.enabled) {
```

Now look at *docs* code you posted:
```dart
if (!state.enabled) {
```

Now read the two **OUT LOUD** to yourself.

**Comment by christocracy** [comment]
Do you know what **`!`** means?

**Comment by mahirozdin** [comment]
👍  of couse I know.. I don't understand how I made a mistake here. I must be very confused. i'm mad for doing this to myself
